src/services/storage-service.js

```diff
@@ -196,16 +196,36 @@ export default class StorageService {
 
     currentDirectory.documentMetadata = getDocumentMetadata({ documents, rooms, cdnObject: currentDirectory });
 
-    objects.forEach(obj => {
+    currentLevelObjects.forEach(obj => {
       obj.documentMetadata = getDocumentMetadata({ documents, rooms, cdnObject: obj });
     });
 
+    if (searchTerm) {
+      const { objects: innerLevelsObjects } = await this._getObjects({
+        parentPath,
+        recursive: true,
+        includeEmptyObjects: false,
+        ignoreNonExistingPath: false
+      });
+
+      const matchingAllLevelsObjects = [
+        ...currentLevelObjects.filter(obj => (
+          obj.documentMetadata?.title.toLowerCase().includes(searchTerm.toLowerCase()) && obj.documentMetadata.isAccessibleToUser)
+          || obj.displayName.toLowerCase().includes(searchTerm.toLowerCase())),
+        ...innerLevelsObjects.filter(obj => obj.displayName.toLowerCase().includes(searchTerm.toLowerCase()))
+      ];
```

Comment:
It took me a while to completely grasp the conditions here, maybe you could extraxct some filter functions, like:

~~~js
const docTitleMatchesSearchTerm = obj => obj.documentMetadata?.title.toLowerCase().includes(searchTerm.toLowerCase()) && obj.documentMetadata.isAccessibleToUser;
const displayNameMatchesSearchTerm = obj => obj.displayName.toLowerCase().includes(searchTerm.toLowerCase());

const matchingAllLevelsObjects = [
  ...currentLevelObjects.filter(obj => docTitleMatchesSearchTerm(obj) || displayNameMatchesSearchTerm(obj)),
  ...innerLevelsObjects.filter(obj => displayNameMatchesSearchTerm(obj))
];
~~~

Reply:
Yes that's indeed much better, I had looked at it too often so that it made sense to me.
Changed!